### PR TITLE
Add arm build to GitHub Actions

### PR DESCRIPTION
**Pull request by asenyaev**
1. Added opencv_contrib repository as a submodule.
2. Created a job for the arm build.
3. Changed the flow of running tests as it is in CI for opencv repository.

### CODE REVIEW

File: .github/workflows/build_wheels_linux_arm.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    env:
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+      REPO_DIR: .
+      BUILD_COMMIT: master
+      PROJECT_SPEC: opencv-python
+      MB_PYTHON_VERSION: ${{ matrix.python-version }}
+      TRAVIS_PYTHON_VERSION: ${{ matrix.python-version }}
+      PLAT: aarch64
+      MB_ML_VER: 2014
+      NP_TEST_DEP: numpy
```
Comment (sergregory) [inline .github/workflows/build_wheels_linux_arm.yml:37]: probably, we should set specific version (some old one?)

File: .github/workflows/build_wheels_linux_arm.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+jobs:
+  build_arm:
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        shell: bash
+
+    strategy:
+      fail-fast: false
```
Comment (sergregory) [inline .github/workflows/build_wheels_linux_arm.yml:19]: why not fail fast?

File: .github/workflows/build_wheels_linux_arm.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    steps:
+    - name: Cleanup
+      if: always()
+      uses: AutoModality/action-clean@v1
+
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
```
Comment (sergregory) [inline .github/workflows/build_wheels_linux_arm.yml:56]: probably should be false?

File: .github/workflows/build_wheels_linux_arm.yml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        fetch-depth: 0
+
+    - name: Setup Environment variables
+      run: |
+        if [ "ARM64" == "${{ matrix.os }}" ]; then echo "PLAT=aarch64" >> $GITHUB_ENV; fi
+        echo "BUILD_DEPENDS=$(echo $NP_BUILD_DEP)" >> $GITHUB_ENV;
+        echo "TEST_DEPENDS=$(echo $NP_TEST_DEP)" >> $GITHUB_ENV;
+
+    - name: before install
```
Comment (sergregory) [inline .github/workflows/build_wheels_linux_arm.yml:65]: General thought: these lines of logic seem to be the same for all linux & OSX builds. Probably, we can extract them to the action file (?) and import it here and there specifying only the needed parameters/env vars.

File: .github/workflows/build_wheels_macos.yml [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+      UNICODE_WIDTH: 32
+      SDIST: ${{ matrix.build_sdist || 0 }}
+      ENABLE_HEADLESS: ${{ matrix.without_gui }}
+      ENABLE_CONTRIB: ${{ matrix.with_contrib }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
```
Comment (sergregory) [inline .github/workflows/build_wheels_macos.yml:50]: submodules should be false

File: .github/workflows/build_wheels_windows.yml [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      SDIST: ${{ matrix.build_sdist || 0 }}
+      ENABLE_HEADLESS: ${{ matrix.without_gui }}
+      ENABLE_CONTRIB: ${{ matrix.with_contrib }}
+      OPENCV_TEST_DATA_PATH: ${{ github.workspace }}\opencv_extra\testdata
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
```
Comment (sergregory) [inline .github/workflows/build_wheels_windows.yml:39]: submodules again

File: .github/workflows/build_wheels_linux_arm.yml (hunk 1 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_linux_arm.yml]: At this moment, python 3.6 supports `numpy <= 1.19.5`, and python 3.9 supports `numpy => 1.19.3`, but there is an issue with `numpy==1.19.5` for the build on ARM. I defined `numpy == 1.19.4` in the latest commit.

File: .github/workflows/build_wheels_linux_arm.yml (hunk 2 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_linux_arm.yml:19]: I wanted to keep it for waiting other builds, to be sure that the fail happened only on a specific build, but not on all builds. In this case, we can keep running a pipeline with several builds, but when only one failed with a network issue or with something else.

File: .github/workflows/build_wheels_linux_arm.yml (hunk 3 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_linux_arm.yml]: Changed it to `false` in the latest commit.

File: .github/workflows/build_wheels_linux_arm.yml (hunk 4 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_linux_arm.yml]: Extracted build and install+test steps into two different scripts in the latest commit, except the macos builds, there is a bit different logic.

File: .github/workflows/build_wheels_macos.yml (hunk 5 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_macos.yml]: Changed it to `false` in the latest commit.

File: .github/workflows/build_wheels_windows.yml (hunk 6 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_windows.yml]: Changed it to `false` in the latest commit.

File: .github/workflows/build_wheels_linux_arm.yml (hunk 1 above)
Comment (sergregory) [reply .github/workflows/build_wheels_linux_arm.yml]: Got it, thanks, I was not aware of this issue. This way we can go with python-specific numpy versions, but it's for the future.

**Review by sergregory — COMMENTED** [review]
LGTM, feel free to merge or ask @asmorkalov when builds will be ready

**Review by sergregory — APPROVED** [review]

File: .github/workflows/build_wheels_linux_arm.yml (hunk 1 above)
Comment (asenyaev) [reply .github/workflows/build_wheels_linux_arm.yml]: I am sorry, I did not link the issue, it is here: https://github.com/opencv/opencv-python/issues/485

**Pull request merged**